platform/engineering/design-docs/2022-06-07-mobile-v2-appointments-index.md

```diff
@@ -0,0 +1,82 @@
+
+## Current Mobile Appointments Functionality
+
+### Index
+
+Because we must maintain older versions of the mobile app, we will need to maintain the V0 appointments index endpoint with its current data schema. [Changes needed are described below.](#backend-adapt-vaos-v2-appointments-to-v0-appointments-schema)
+
+### Create
+
+The create appointment endpoint only works for appointment requests at this time and already uses the V2 API. This endpoint will not be in use until after the switch to V2. No action is needed here.
+
+### Cancel
+
+The cancel endpoint is currently on V0 for both appointments and appointment requests. [This will need to be updated to V2.](#backend-v2-appointment-cancel)
+
+## Web App V2 Appointments Index Functionality
+
+The web app appointments V2 index makes three different requests, depending on which drop-down option the user selects:
+  - Upcoming: requests booked, arrived, fulfilled, and cancelled appointments for date range of one month ago through 13 months from now.
+    - This call is always made, even if the user bypasses the upcoming appointments and navigates directly to another appointment type.
+    - Example (recorded on 2022-06-08):
+      https://staging-api.va.gov/vaos/v2/appointments?_include=facilities,clinics&start=2022-05-09&end=2023-07-08&statuses[]=booked&statuses[]=arrived&statuses[]=fulfilled&statuses[]=cancelled
+  - Past: requests booked, arrived, fulfilled, and cancelled appointments for the past three months until the current time.
+    - Example (recorded on 2022-06-08):
+	    https://staging-api.va.gov/vaos/v2/appointments?_include=facilities,clinics&start=2022-03-08T00:00:00-05:00&end=2022-06-08T16:47:08-04:00&statuses[]=booked&statuses[]=arrived&statuses[]=fulfilled&statuses[]=cancelled
+  - Cancelled and Requested: requests cancelled and proposed appointments for the past four months through today.
+    - When the user navigates to cancelled appointments on the web app, it only fetches the past four months of cancelled appointments. However, when the user navigates to cancelled, it always makes the "upcoming" fetch that includes cancelled appointments from one month ago to 13 months from now before fetching the past four months of cancelled appointments. This is true even if the user pastes in the url to navigate directly to cancelled appointments without going to any other appointments page first. Then it only seems to show cancelled appointments starting from one month ago. We should look at front end code to better understand why this is happening. We may not need to request cancelled appointments from more than one month ago if the web app doesn't show them.
+    - Example (recorded on 2022-06-08):
+    	https://staging-api.va.gov/vaos/v2/appointments?_include=facilities,clinics&start=2022-02-08&end=2022-06-08&statuses[]=proposed&statuses[]=cancelled
+
+## Tickets Needed
+
+### Backend: Adapt VAOS V2 Appointments to V0 Appointments Schema
+
+This will involve:
```

Comment:
I'd add a ticket just for coordinating the switch over and have some steps listed like: 1. switch feature flag 2. run integration tests 3. check datadog etc. Otherwise looks good!